fix: make log transfer operational

doudou opened the pull request:
Depends on:
 - [ ] https://github.com/tidewise/bundles-wetpaint/pull/257

These PR adds rate limiting, which was a needed feature. The rest is to make the setup of it easier, and have a way to transfer data even for localhost (great to test, and will be useful since we want to have a dedicated log server)

Commit:
chore(net_gen): remove extra ')' from a warn message

doudou committed:
feat: implement rate-liming in log transfer

doudou committed:
fix: cleanup how one should configure the log upload functionality

doudou committed:
fix: switch log upload to implicit FTPs

Meaning, both the server and client start in TLS mode. Without this,
there is some exchange "in the clear" before they switch.

I made this change not for the sake of security, but because the explicit
mode started failing for no discernable reason.

doudou committed:
fix: raise an error if log transfer is enabled but IP is not set

doudou committed:
fix: flush log transfers on quit

doudou committed:
feat: handle configuration of the max upload rates

doudou committed:
chore: separate most of the log transfer's app integration code from Plugin

For better organization and testing

doudou committed:
fix: ack the result of remote log directory creation command

doudou committed:
chore: move log result struct under LogUploadState

doudou committed:
fix: only use ProcessServerConfig to manipulate process servers in log transfer code

doudou committed:
fix: log transfer integration within Plugin

doudou committed:
fix: make loggers per-server, and inhibit log messages in tests

doudou committed:
fix: inhibit last log messages in test_log_transfer_manager

doudou committed:
fix: type mismatch in argument passed to LogTransferManager#transfer

doudou committed:
fix: set supports_log_transfer on the local process server

caioaamaral commented on lib/syskit/network_generation/logger.rb:
```diff
@@ -129,7 +129,7 @@ def self.add_logging_to_network(engine, work_plan)
                         warn "deployment #{deployment.process_name} has no usable "\
                              "logger (default logger name would be "\
-                             "#{deployment.process_name}_Logger))"
+                             "#{deployment.process_name}_Logger)"
```
I had fixed that, but my target was on master instead of 'transition-to-runkit' https://github.com/rock-core/tools-syskit/pull/336

every PR on tools-syskit (from our side) should target `transition-to-runkit`?

doudou replied:
Until we get the runkit branch merged, yes, I'm afraid.

It should merge cleanly, so I don't think it's an issue.

doudou committed:
fix: implicit FTPS usage should be default only after 2.7.0

This workarounds some incompatibility between net-ftp and ftpd. They
don't manage connecting properly in implicit mode before 2.7.0, and
don't manage connecting properly in explicit mode afterwards

doudou committed:
fix: trivial unit test fix

doudou committed:
fix: rubocop offenses
